### DISCUSSION TITLE
Is there any interest in more life cycle hooks for traits?

### DISCUSSION THREAD

**Discussion by xxdalexx** [livewire/livewire#2396] [3. Feature ideas]
I know it was originally planned that "plugins" could be done through traits, and for those that don't know (because it's undocumented) there's already very limited functionality for it.

If you put a method in your trait `initializeTraitName()` it gets called at the beginning of every request lifecycle, before mount/hydrate.

I reach for these on a semi regular basis, and I've come up with some hacky ways of dealing with the mount vs hydrate, or hard coding in some calls to trait methods in the regular lifecycle methods you would put into your component.

I'd be interested in submitting a PR that would take the idea of the api from the current single method available, and change it to include `mountTraitName()`, `hydrateTraitName()`, and `dehydrateTraitName()`.

**Comment by joshhanley** [comment]
@xxdalexx have a look at the traits docs page, at the bottom, there is a big code block that has all of those methods your asking for https://laravel-livewire.com/docs/2.x/traits

Hope this helps!

![image](https://user-images.githubusercontent.com/882837/105663990-a87c1700-5f27-11eb-9ebc-c83fbcefa8aa.png)

**Reply by xxdalexx** [reply]
Thank you, I don't know how I missed this.
